src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/ShopUserFactory.php

```diff
@@ -0,0 +1,153 @@
+ * @method static ShopUserInterface[]|Proxy[] randomSet(int $number, array $attributes = [])
+ * @method static ShopUserInterface[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
+ * @method ShopUserInterface|Proxy create(array|callable $attributes = [])
+ */
+final class ShopUserFactory extends ModelFactory implements ShopUserFactoryInterface
+{
+    public function __construct(
+        private FactoryInterface $shopUserFactory,
+        private FactoryInterface $customerFactory,
+        private CustomerGroupFactoryInterface $customerGroupFactory,
+    ) {
+        parent::__construct();
+    }
+
+    public function withEmail(string $email): self
+    {
+        return $this->addState(['email' => $email]);
+    }
+
+    public function withFirstName(string $firstName): self
+    {
+        return $this->addState(['first_name' => $firstName]);
+    }
+
+    public function withLastName(string $lastName): self
+    {
+        return $this->addState(['last_name' => $lastName]);
+    }
+
+    public function male(): self
+    {
+        return $this->addState(['gender' => CustomerInterface::MALE_GENDER]);
+    }
+
+    public function female(): self
+    {
+        return $this->addState(['gender' => CustomerInterface::FEMALE_GENDER]);
+    }
+
+    public function withPhoneNumber(string $phoneNumber): self
+    {
+        return $this->addState(['phone_number' => $phoneNumber]);
+    }
+
+    public function withBirthday(\DateTimeInterface|string $birthday): self
+    {
+        return $this->addState(function () use ($birthday): array {
+            if (is_string($birthday)) {
+                return ['birthday' => new \DateTimeImmutable($birthday)];
+            }
+
+            return ['birthday' => $birthday];
+        });
+    }
+
+    public function withPassword(string $password): self
+    {
+        return $this->addState(['password' => $password]);
+    }
+
+    public function withCustomerGroup(Proxy|CustomerGroupInterface|string $customerGroup): self
+    {
+        return $this->addState(function () use ($customerGroup): array {
+            if (is_string($customerGroup)) {
+                return ['customer_group' => $this->customerGroupFactory::randomOrCreate(['code' => $customerGroup])];
+            }
+
+            return ['customer_group' => $customerGroup];
+        });
+    }
+
+    protected function getDefaults(): array
+    {
+        return [
+            'email' => self::faker()->email(),
+            'first_name' => self::faker()->firstName(),
+            'last_name' => self::faker()->lastName(),
+            'enabled' => true,
+            'password' => 'password123',
+            'customer_group' => $this->customerGroupFactory::randomOrCreate(),
+            'gender' => CustomerInterface::UNKNOWN_GENDER,
+            'phone_number' => self::faker()->phoneNumber(),
+            'birthday' => self::faker()->dateTimeThisCentury(),
```

Comment:
Very young audience. How about generating them of ages between 18 and 80 years: `->dateTimeBetween('-80 years', '-18 years')`?

Reply:
cause it's the current implementation, but you're right, it makes sense. @lchrusciel @Zales0123 What do you think?

https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Fixture/Factory/ShopUserExampleFactory.php#L87

Reply:
👍 for the change, especially in the new implementation. In the current one, you can open a PR and we will see ;)

Reply:
@vvasiloi done